GenericsHomework/NodeCollection.cs

```diff
@@ -0,0 +1,137 @@
+﻿using System.Collections;
+
+namespace GenericsHomework;
+public class NodeCollection<T> : IEnumerator<T>, ICollection<T> where T : notnull
```

Comment:
```suggestion
public class NodeCollection<T> : ICollection<T> where T : notnull
```
You should not implement `IEnumerator<T>`. It is not thread safe